src/services/tasks.service.test.ts

```diff
@@ -1,72 +1,190 @@
-//     await models.task.sync({force: true});
-
-//     await models.task.destroy({
-//         truncate: true
-//     });
-
-//     await models.task.bulkCreate(testTasks);
-// });
-
-// afterAll(()=>{
-//     database.close();
-// });
-
-// describe('get_tasks', ()=> {
-//     test('Should return only incomplete', async() => {        
-//         const tasks = await _.get_tasks('incomplete'); 
-//         expect(tasks).toHaveLength(3);
-
-//         tasks.forEach((task)=>{
-//             expect(task.get('status')).toBe('incomplete');
-//             expect(testTasks).toContainEqual({task: task.get('task'), status: task.get('status')});
-//         })
-//     });
-
-//     test('Should return only complete', async() => {
-
-//         const tasks = await _.get_tasks('complete'); 
-//         expect(tasks).toHaveLength(1);
-
-//         tasks.forEach((task)=>{
-//             expect(task.get('status')).toBe('complete');
-//             expect(testTasks).toContainEqual({task: task.get('task'), status: task.get('status')});
-//         })
-
-//     });
-// });
-
-// describe('get_all_tasks', ()=> {
-//     test('Should return all tasks in the database', async() => {    
-//         const tasks = await _.get_all_tasks(); 
-//         expect(tasks).toHaveLength(4);
-
-//         tasks.forEach((task)=>{
-//             expect(testTasks).toContainEqual({task: task.get('task'), status: task.get('status')});
-//         })
-
-//     });    
-// })
+import { taskService } from '.';
+import { Task } from '../models/task.model';
+import { sequelize } from './database';
+import { ETaskStatus } from '../models/share';
+import '../config';
+
+const testTasks = [
+  {task: 'Buy eggs',
+  status: ETaskStatus.Incomplete},
+  {task: 'Buy milk',
+  status: ETaskStatus.Complete},
+  {task: 'Buy cheese',
+  status: ETaskStatus.Incomplete},
+  {task: 'Buy bacon',
+  status: ETaskStatus.Incomplete},
+];
+
+const newTask = 'Buy chocolate';
+
+// Known Jest async bug:
+// https://github.com/facebook/jest/issues/6619
+// Setting longer timeout to encourage tests to work.
+// Sometimes tests will fail due to async issues. Running again makes them work.
+
+jest.setTimeout(10000);
+
+const instantiateDB = async () => {
+  await Task.sync();
+  await Task.destroy({
+      truncate: true
+  });
+  await Task.bulkCreate(testTasks);
+}
+
+beforeEach(async ()=>{
+  await instantiateDB();
+});
+
+afterAll(()=>{
+  sequelize.close();
+});
+
+describe('getTaskByStatus', ()=> {
+
+  test('Should return only incomplete', async() => {        
+    const tasks = await taskService.getTaskByStatus(ETaskStatus.Incomplete); 
+    expect(tasks).toHaveLength(3)
+    tasks.forEach((task)=>{
+      expect(task.status).toBe(ETaskStatus.Incomplete);
+      expect(testTasks).toContainEqual({task: task.task, status: task.status});
+    });
+  });
+
+  test('Should return only complete', async() => {
+    const tasks = await taskService.getTaskByStatus(ETaskStatus.Complete); 
```

Comment:
If you can refactor the code here, that will be much better. This test and the above one only different from the task status. Basically, we can write a function, taking an generic arg